src/cmd-buildupload

```diff
@@ -166,13 +171,16 @@ def s3_upload_build(args, builddir, bucket, prefix):
         f.flush()
         s3_copy(f.name, bucket, f'{prefix}/meta.json',
             CACHE_MAX_AGE_METADATA, args.acl,
-            dry_run=args.dry_run)
+            dry_run=args.dry_run, endpoint_url=args.endpoint_url)
 
 
 @retry(stop=retry_stop, retry=retry_boto_exception, before_sleep=retry_callback)
-def s3_check_exists(bucket, key, dry_run=False):
+def s3_check_exists(bucket, key, dry_run=False, endpoint_url=None):
     print(f"Checking if bucket '{bucket}' has key '{key}'")
-    s3 = boto3.client('s3')
+    if endpoint_url is not None:
+        s3 = boto3.client('s3', endpoint_url=endpoint_url)
+    else:
+        s3 = boto3.client('s3')
```

Comment:
The default `endpoint_url` value for `client()` is also `None`, so this can be collapsed to

```suggestion
    s3 = boto3.client('s3', endpoint_url=endpoint_url)
```